api/envoy/api/v2/eds.proto

```diff
@@ -80,6 +81,12 @@ message ClusterLoadAssignment {
     //    "lb"_drop = 20%  // 50% of the remaining 'actual' load, which is 40%.
     //    actual_outgoing_load = 20% // remaining after applying all categories.
     repeated DropOverload drop_overloads = 2;
+
+    // Priority levels and localities are considered overprovisioned with this
+    // factor (in percentage). This means that we don't consider a priority
+    // level or locality unhealthy until the percentage of healthy hosts
+    // multiplied by the over provisioning factor drops below 100.
+    google.protobuf.UInt32Value over_provisioning_factor = 3 [(validate.rules).uint32.gt = 0];
```

Comment:
Because the math here is a bit confusing, it might be worth linking to load_balancing.rst here

We'll also want to update the docs as part of this PR to note that you can change the default level now that it's not hard-coded any more :-)

https://www.envoyproxy.io/docs/envoy/v1.7.0/intro/arch_overview/load_balancing.html?highlight=priorities#priority-levels